pina/model/layers/embedding.py

```diff
@@ -0,0 +1,142 @@
+""" Embedding modulus. """
+
+import torch
+from pina.utils import check_consistency
+
+
+class PBCEmbedding(torch.nn.Module):
+    r"""
+    Imposing hard constraint periodic boundary conditions by embedding the
+    input. 
+    
+    A periodic function :math:`u:\mathbb{R}^{\rm{in}}
+    \rightarrow\mathbb{R}^{\rm{out}}` periodic in the spatial
+    coordinates :math:`\mathbf{x}` with periods :math:`\mathbf{L}` is such that:
+
+    .. math::
+        u(\mathbf{x})  = u(\mathbf{x} + n \mathbf{L})\;\;
+        \forall n\in\mathbb{N}.
+
+    The :meth:`PBCEmbedding` augments the input such that the periodic conditons
+    is guarantee. The input is augmented by the following formula:
+
+    .. math::
+        \mathbf{x} \rightarrow \tilde{\mathbf{x}} = \left[1,
+        \cos\left(\frac{2\pi}{L_1} x_1 \right),
+        \sin\left(\frac{2\pi}{L_1}x_1\right), \cdots,
+        \cos\left(\frac{2\pi}{L_{\rm{in}}}x_{\rm{in}}\right),
+        \sin\left(\frac{2\pi}{L_{\rm{in}}}x_{\rm{in}}\right)\right],
+
+    where :math:`\text{dim}(\tilde{\mathbf{x}}) = 3\text{dim}(\mathbf{x})`.
+
+    .. seealso::
+        **Original reference**: 
+            1.  Dong, Suchuan, and Naxian Ni (2021). *A method for representing
+                periodic functions and enforcing exactly periodic boundary
+                conditions with deep neural networks*. Journal of Computational
+                Physics 435, 110242.
+                DOI: `10.1016/j.jcp.2021.110242.
+                <https://doi.org/10.1016/j.jcp.2021.110242>`_
+            2.  Wang, S., Sankaran, S., Wang, H., & Perdikaris, P. (2023). *An
+                expert's guide to training physics-informed neural networks*.
+                DOI: `arXiv preprint arXiv:2308.0846.
+                <https://arxiv.org/abs/2308.08468>`_
+    .. warning::
+        The embedding is a truncated fourier expansion, and only ensures
+        function PBC and not for its derivatives. Ensuring approximate
+        periodicity in
+        the derivatives of :math:`u` can be done, and extensive
+        tests have shown (also in the reference papers) that this implementation
+        can correctly compute the PBC on the derivatives up to the order
+        :math:`\sim 2,3`, while it is not guarantee the periodicity for
+        :math:`>3`. The PINA code is tested only for function PBC and not for
+        its derivatives.
```

Comment:
A small example to show the minimal usage

Reply:
ok I will add it.